### PR TITLE
Xml refactor

### PR DESCRIPTION
**Pull request by codymccoy**

### CODE REVIEW

**Comment by rnugent3** [comment]
Is the unit test no longer relevant?

**Comment by codymccoy** [comment]
As it turns out, you can't run a unit test that calls into a method that uses MessageBox without it throwing an exception. "BadImageException" if i remember correctly. I looked into how to solve this and there didn't seem to be a simple solution. I would like the message box to be in the method that it is in. I didn't think it was worth the hassle to figure this one out so i commented it out for now. I wrote the reason why above the commented out test.

**Review by Brennan1994 — CHANGES_REQUESTED** [review]

File: HEC.FDA.ViewModelTest/WriteAndReadToXMLShould.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            XElement elemXML = elem1.ToXML();
+
+            LateralStructureElement elem2 = new LateralStructureElement(elemXML, id);
+
+            Assert.True(elem1.Equals(elem2));
+        }
+
+        //this test was failing because it calls code that has a message box.
+        //[Fact]
+        //public void TestStageDamageElementWriteThenReadAreEqual()
```
[the remark this answers is not in the document]
Comment (codymccoy) [reply HEC.FDA.ViewModelTest/WriteAndReadToXMLShould.cs:175]: In my occtypes_to_xml branch I now have this test working. After working through it, there was a bug between the write and the read from xml.

File: ViewModel/HEC.FDA.ViewModel.csproj [hunk 2]
```diff
@@ -22,6 +22,20 @@
 		<Compile Remove="ImpactAreaScenario\Results\DamageByDamageCategoryVM.cs" />
 	</ItemGroup>
 
+  <ItemGroup>
```
[the remark this answers is not in the document]
Comment (codymccoy) [reply ViewModel/HEC.FDA.ViewModel.csproj:25]: done

File: HEC.FDA.ViewModelTest/WriteAndReadToXMLShould.cs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+using HEC.FDA.ViewModel.IndexPoints;
+using HEC.FDA.ViewModel.StageTransforms;
+using HEC.FDA.ViewModel.TableWithPlot;
+using HEC.FDA.ViewModel.Utilities;
+using HEC.FDA.ViewModel.Watershed;
+using System.Collections.Generic;
+using System.Xml.Linq;
+using Xunit;
+
+namespace HEC.FDA.ViewModelTest
```
[the remark this answers is not in the document]
Comment (codymccoy) [reply HEC.FDA.ViewModelTest/WriteAndReadToXMLShould.cs:17]: done

[the remark this answers is not in the document]
Comment (codymccoy) [reply HEC.FDA.ViewModelTest/WriteAndReadToXMLShould.cs:17]: I created new classes for the unit tests. The last one that was commented out will be in its own class on my latest branch.

**Pull request merged**